app/main.py

```diff
@@ -1,3 +1,60 @@
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union["Distance", float]) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: Union["Distance", float]) -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, factor: float) -> "Distance":
+        return Distance(self.km * factor)
+
+    def __truediv__(self, divisor: float) -> "Distance":
+        return Distance(round(self.km / divisor, 2))
```

Comment:
There is a potential issue with division by zero in the `__truediv__` method. Consider adding a check to ensure that the `divisor` is not zero before performing the division to prevent runtime errors.